### PR TITLE
Add pagination to activities table

### PR DESCRIPTION
**Pull request by yksolanki9**
Fixes  https://github.com/ghostfolio/ghostfolio/issues/1354
- Added pagination to the `activities-table` component
- Pagination toolbar is hidden if pagination is disabled(page size is larger than number of rows) or if there is only one page or if user has no activity
- The activity count, total value and total fees calculation changes according to the page the user is on
- Also fixed some existing formatting errors in the code

UI Preview:
Screenshot 2022-11-04 at 12 10 34 AM

### CODE REVIEW

File: libs/ui/src/lib/activities-table/activities-table.component.ts [hunk 1]
```diff
@@ -37,6 +38,7 @@ export class ActivitiesTableComponent implements OnChanges, OnDestroy {
   @Input() hasPermissionToImportActivities: boolean;
   @Input() hasPermissionToOpenDetails = true;
   @Input() locale: string;
+  @Input() pageSize = Number.MAX_SAFE_INTEGER;
```
Comment (dtslvr) [inline libs/ui/src/lib/activities-table/activities-table.component.ts:41]: I would use a default page size of 50 for every table:

```
@Input() pageSize = DEFAULT_PAGE_SIZE;
```

Add `DEFAULT_PAGE_SIZE=50` to `libs/common/src/lib/config.ts`

File: apps/client/src/app/pages/portfolio/transactions/transactions-page.html [hunk 2]
```diff
@@ -3,6 +3,7 @@
     <div class="col">
       <h3 class="d-flex justify-content-center mb-3" i18n>Activities</h3>
       <gf-activities-table
+        pageSize="50"
```
Comment (dtslvr) [inline apps/client/src/app/pages/portfolio/transactions/transactions-page.html:6]: Remove line (see comment in `libs/ui/src/lib/activities-table/activities-table.component.ts`)

File: libs/ui/src/lib/activities-table/activities-table.component.ts [hunk 3]
```diff
@@ -273,6 +293,13 @@ export class ActivitiesTableComponent implements OnChanges, OnDestroy {
     return totalValue.toNumber();
   }
 
+  public pageChanged(page: PageEvent) {
```
Comment (dtslvr) [inline libs/ui/src/lib/activities-table/activities-table.component.ts:296]: Move up to the other `public` methods

**Review by dtslvr — CHANGES_REQUESTED** [review]
Can you please add

```
.mat-paginator {
  background-color: rgba(var(--palette-foreground-base-dark), 0.02);
}
```

to `apps/client/src/styles.scss` (above `.svgMap-tooltip {`)

and this

```
.mat-paginator {
  background-color: rgba(var(--palette-foreground-base-light), 0.02);
  
  .mat-paginator-page-size {
    display: none;
  }
}
```

above `.no-min-width {`

File: libs/ui/src/lib/activities-table/activities-table.component.ts (hunk 1 above)
Comment (yksolanki9) [reply libs/ui/src/lib/activities-table/activities-table.component.ts:41]: Fixed

File: libs/ui/src/lib/activities-table/activities-table.component.ts (hunk 3 above)
Comment (yksolanki9) [reply libs/ui/src/lib/activities-table/activities-table.component.ts:296]: Done

File: apps/client/src/app/pages/portfolio/transactions/transactions-page.html (hunk 2 above)
Comment (yksolanki9) [reply apps/client/src/app/pages/portfolio/transactions/transactions-page.html:6]: Done

**Review by dtslvr — APPROVED** [review]
Tip-top! ✨  Thank you @yksolanki9 🤓

**Pull request merged**